validate-workflow.sh

```
@@ -0,0 +1,96 @@
+# Check if working tree is clean
+if [ -n "$(git status --porcelain)" ]; then
+    echo "❌ Working tree is not clean. Please commit or stash changes"
+    git status --short
+    exit 1
+fi
+
+# Check if branch is up to date with master
+if [ "$CURRENT_BRANCH" != "master" ]; then
+    echo "🔄 Checking if feature branch is up to date with master..."
+    git fetch origin master
+    BEHIND=$(git rev-list --count master..$CURRENT_BRANCH)
```

Shellcheck (0.11.0)

[warning] BEHIND appears unused. Verify use (or export if used externally).

(SC2034)